File: multiCamCommon/java/src/main/java/com/multicam/common/CommandType.java
package com.multicam.common;

import com.google.gson.annotations.SerializedName;

/**
 * Available command types for the MultiCam API.
 */
public enum CommandType {
    /** Start video recording (immediate or scheduled) */
    @SerializedName("START_RECORDING")
    START_RECORDING,

    /** Stop current recording and return file ID */
    @SerializedName("STOP_RECORDING")
    STOP_RECORDING,

    /** Query current device status */
    @SerializedName("DEVICE_STATUS")
    DEVICE_STATUS,

    /** Download video file (binary protocol) */
    @SerializedName("GET_VIDEO")
    GET_VIDEO,

    /** Health check ping */
    @SerializedName("HEARTBEAT")
    HEARTBEAT,

    /** List available video files (may not be supported on all platforms) */
    @SerializedName("LIST_FILES")
    LIST_FILES,

    /** Upload video file to cloud using presigned S3 URL */
    @SerializedName("UPLOAD_TO_CLOUD")
    UPLOAD_TO_CLOUD
}

File: multiCamCommon/swift/Sources/MultiCamCommon/CommandType.swift
import Foundation

/// Available command types for the MultiCam API
public enum CommandType: String, Codable, CaseIterable {
    /// Start video recording (immediate or scheduled)
    case startRecording = "START_RECORDING"

    /// Stop current recording and return file ID
    case stopRecording = "STOP_RECORDING"

    /// Query current device status
    case deviceStatus = "DEVICE_STATUS"

    /// Download video file (binary protocol)
    case getVideo = "GET_VIDEO"

    /// Health check ping
    case heartbeat = "HEARTBEAT"

    /// List available video files (may not be supported on all platforms)
    case listFiles = "LIST_FILES"

    /// Upload video file to cloud using presigned S3 URL
    case uploadToCloud = "UPLOAD_TO_CLOUD"
}
